Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -284,16 +287,147 @@ class NexusGeometrySaveTest : public CxxTest::TestSuite {
 
-    // expectedInstrumentName; // let dataset name have same value as component
-    // name.
+    std::string expectedDataSetValue = "test_data_for_instrument";
 
     TS_ASSERT(testUtility.hasDataSet(expectedInstrumentName,
+                                     expectedDataSetValue,
                                      "/raw_data_1/instrument"));
 
     TS_ASSERT(testUtility.hasAttributeInDataSet(
         expectedInstrumentName, "/raw_data_1/instrument", SHORT_NAME,
         expectedInstrumentName));
   }
+
+  void test_translation() {
+
+
+    auto instrument =
+        ComponentCreationHelper::createTestInstrumentRectangular2(1, 10);
+
+    auto beamline =
+        Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+    auto &compInfo = (*beamline.first);
+
+    // translate source
+    Eigen::Vector3d absoluteBankLocation(5, 0, 0);
+
+    compInfo.setPosition(compInfo.indexOfAny("bank1"),
+                         Mantid::Kernel::toV3D(absoluteBankLocation));
+
+    auto relativeSourcePosition = Mantid::Kernel::toVector3d(
+        compInfo.relativePosition(compInfo.indexOfAny("bank1")));
+
+    TS_ASSERT(relativeSourcePosition.isApprox(absoluteBankLocation));
+  }
+
+
+
+  void test_rotation() {
+
+    auto instrument =
+        ComponentCreationHelper::createTestInstrumentRectangular2(1, 10);
+
+    auto beamline =
+        Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+    auto &compInfo = (*beamline.first);
+
+    Eigen::Quaterniond absoluteBankRotation(
+        Eigen::AngleAxisd(M_PI / 4, Eigen::Vector3d(0, 1, 0)));
+
+    compInfo.setRotation(
+        compInfo.indexOfAny("bank1"), // bank1 defined as in
+                                      // ComponentCreationHelper.cpp
+                                      // line 584 01.07.19
+        Mantid::Kernel::toQuat(absoluteBankRotation));
+
+    auto relativeBankRotation = Mantid::Kernel::toQuaterniond(
+        compInfo.relativeRotation(compInfo.indexOfAny("bank1")));
+
+    TS_ASSERT(relativeBankRotation.isApprox(absoluteBankRotation));
+  }
+
+  void
+  test_bank_absolute_rotation_is_equal_to_detector_in_bank_absolute_rotation() {
+
+    auto instrument =
+        ComponentCreationHelper::createTestInstrumentRectangular2(1, 10);
+
+    auto beamline =
+        Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+    auto &compInfo = (*beamline.first);
+
+    // set abosolute bank rotation (rotate about y)
+    Eigen::Quaterniond absoluteBankRotation(
+        Eigen::AngleAxisd(M_PI / 4, Eigen::Vector3d(0, 1, 0)));
+
+    Eigen::Quaterniond zeroRotation(
+        Eigen::AngleAxisd(0, Eigen::Vector3d(0, 0, 0)));
+
+    // relative position of detector
+    auto relativeDetectorRotation =
+        Mantid::Kernel::toQuaterniond(compInfo.relativeRotation(0));
+
+    TS_ASSERT(relativeDetectorRotation.isApprox(zeroRotation));
+  }
+
+
+  void test_detector_position_is_expected_value_after_rotation() {
+
+    // 45 degree rotation around z.
```

Comment:
Put a longer comment here to explain the setup or even add some ascii art to show setup.

Reply:
Do similar for all test setups that would need explaining for yourself or future developers.